From 3a901d16ebccb3c44721cb4e4701977714188b93 Mon Sep 17 00:00:00 2001
From: Sandeep Sheriker M <sandeep.sheriker@microchip.com>
Date: Mon, 3 Jun 2019 17:14:36 -0700
Subject: [gst1-hantro-g1][patch  1/5] gst1-hantro-g1: configure drm/kms cflags
 & ldflags

set drm/kms cflags & ldflags through autotools instead of setting it
in Makefile.am through pkg-config.

Signed-off-by: Sandeep Sheriker M <sandeep.sheriker@microchip.com>
---
 configure.ac            | 7 +++++++
 sys/drmsink/Makefile.am | 8 +++-----
 sys/kms/Makefile.am     | 8 +-------
 3 files changed, 11 insertions(+), 12 deletions(-)

diff --git a/configure.ac b/configure.ac
index 041e374..ad4854c 100644
--- a/configure.ac
+++ b/configure.ac
@@ -240,6 +240,12 @@ dnl these are all the gst plug-ins, compilable without additional libs
 dnl AG_GST_CHECK_PLUGIN(pluginname)
 
 dnl *** sys plug-ins ***
+dnl *** kms ***
+translit(dnm, m, l) AM_CONDITIONAL(USE_KMS, true)
+AG_GST_CHECK_FEATURE(KMS, [drm/kms libraries], kms, [
+  AG_GST_PKG_CHECK_MODULES(GST_ALLOCATORS, gstreamer-allocators-1.0)
+  PKG_CHECK_MODULES([KMS_DRM], [libdrm >= 2.4.55], HAVE_KMS=yes, HAVE_KMS=no)
+])
 
 dnl *** ext plug-ins ***
 dnl keep this list sorted alphabetically !
@@ -261,6 +267,7 @@ else
 dnl not building plugins with external dependencies,
 dnl but we still need to set the conditionals
 AM_CONDITIONAL(USE_G1, false)
+AM_CONDITIONAL(USE_KMS, false)
 
 fi dnl of EXT plugins
 
diff --git a/sys/drmsink/Makefile.am b/sys/drmsink/Makefile.am
index 09ad4b3..63b8c25 100644
--- a/sys/drmsink/Makefile.am
+++ b/sys/drmsink/Makefile.am
@@ -1,20 +1,18 @@
 plugin_LTLIBRARIES = libgstdrmsink.la
-LIBDRM_CFLAGS = `pkg-config --cflags libdrm --cflags libkms`
 libgstdrmsink_la_SOURCES = gstdrmsink.c gstframebuffersink.c
 libgstdrmsink_la_CFLAGS = \
         $(GST_PLUGINS_BASE_CFLAGS) \
         $(GST_CFLAGS) \
-        $(LIBFBDEV_CFLAGS) \
-        $(LIBDRM_CFLAGS) \
+        $(KMS_DRM_CFLAGS) \
        -I$(top_builddir)/gst-libs/ext/g1/memalloc \
        -I$(top_builddir)/gst-libs/ext/g1/bus
                  
 libgstdrmsink_la_LIBADD = \
         $(GST_PLUGINS_BASE_LIBS) -lgstvideo-$(GST_API_VERSION) -lrt \
         $(GST_BASE_LIBS) \
-        $(LIBFBDEV_LIBS) \
+        $(KMS_DRM_LIBS) \
         $(top_builddir)/gst-libs/ext/g1/bus/libgstbusallocator-@GST_API_VERSION@.la \
-        $(GST_LIBS) -ldrm -lkms 
+        $(GST_LIBS)
               
 libgstdrmsink_la_LDFLAGS = $(GST_PLUGIN_LDFLAGS)
 libgstdrmsink_la_LIBTOOLFLAGS = $(GST_PLUGIN_LIBTOOLFLAGS) 
diff --git a/sys/kms/Makefile.am b/sys/kms/Makefile.am
index 3bd0891..73c093e 100644
--- a/sys/kms/Makefile.am
+++ b/sys/kms/Makefile.am
@@ -1,5 +1,4 @@
 plugin_LTLIBRARIES = libgstg1kms.la
-LIBDRM_CFLAGS = `pkg-config --cflags libdrm --cflags libkms`
 libgstg1kms_la_SOURCES = 			\
 	gstg1kmssink.c				\
 	gstkmsutils.c				\
@@ -14,7 +13,6 @@ libgstg1kms_la_CFLAGS = 			\
 	$(GST_ALLOCATORS_CFLAGS)		\
 	$(GST_CFLAGS) 				\
 	$(KMS_DRM_CFLAGS) 			\
-	$(LIBDRM_CFLAGS)			\
        -I$(top_builddir)/gst-libs/ext/g1/memalloc \
        -I$(top_builddir)/gst-libs/ext/g1/bus \
 	$(NULL)
@@ -25,7 +23,7 @@ libgstg1kms_la_LIBADD = 			\
 	$(GST_VIDEO_LIBS)			\
 	$(GST_ALLOCATORS_LIBS)			\
 	$(GST_LIBS) 				\
-	$(KMS_DRM_LIBS) -ldrm -lkms		\
+	$(KMS_DRM_LIBS)		\
 	-lgstvideo-$(GST_API_VERSION) -lrt	\
 	-lgstallocators-$(GST_API_VERSION)	\
 	$(top_builddir)/gst-libs/ext/g1/bus/libgstbusallocator-@GST_API_VERSION@.la \
@@ -35,10 +33,6 @@ libgstg1kms_la_LDFLAGS = 			\
 	$(GST_PLUGIN_LDFLAGS)			\
 	$(NULL)
 
-libgstg1kms_la_LIBTOOLFLAGS = 		\
-	$(GST_PLUGIN_LIBTOOLFLAGS) 		\
-	$(NULL)
-
 noinst_HEADERS = 				\
 	gstg1kmssink.h 				\
 	gstkmsutils.h				\
-- 
2.7.4

